packages/dendron-plugin-views/src/hooks/useGraphElements.tsx

```diff
@@ -34,166 +35,246 @@ const DEFAULT_CLASSES = ""; //graph-view
 const DEFAULT_NODE_CLASSES = `${DEFAULT_CLASSES}`; //color-fill
 const DEFAULT_EDGE_CLASSES = `${DEFAULT_CLASSES}`;
 
-const getLocalNoteGraphElements = ({
+type QueueData = {
+  note: NoteProps;
+  /**
+   * How far (in the graph) away is this note from the original note?
+   */
+  distance: number;
+  classes: string;
+  /**
+   * is node an ancestor of the active note
+   */
+  isParent?: boolean;
+};
+
+function computeGraphElements({
```

Comment:
Gave the graph view a test, and I don't quite recall what we had discussed is the functionally right thing to do, but I _think_ the view will be more useful if we treat hierarchies and links as different types of nodes - meaning that if we set depth = 2, then we'd include the following nodes: 

- parent, grandparent
- children, grandchildren
- active note's links
- active note's links' links

In this proposal, things we will **not** include are:
- siblings, cousins, aunts, uncles, nieces, nephews, etc. If a user wants to view siblings, then they can navigate to the parent note instead (and the sibling will show up as the parent's children)
- parent's links, grandparent's links, childrens' links, etc.
- a link's parent/child. 

What do you think about this proposal?

Reply:
updated.